riscv/src/riscv_gl.asm

```diff
@@ -0,0 +1,345 @@
+{{IMPORTS}}
+
+let MIN_DEGREE_LOG: int = {{MIN_DEGREE_LOG}};
+let MIN_DEGREE: int = 2**MIN_DEGREE_LOG;
+let MAX_DEGREE_LOG: int = {{MAX_DEGREE_LOG}};
+let MAIN_MAX_DEGREE: int = 2**MAX_DEGREE_LOG;
+let LARGE_SUBMACHINES_MAX_DEGREE: int = 2**(MAX_DEGREE_LOG + 2);
+
+machine Main with min_degree: MIN_DEGREE, max_degree: {{MAIN_MAX_DEGREE}} {
+
+    reg pc[@pc];
+    reg X[<=];
+    reg Y[<=];
+    reg Z[<=];
+    reg W[<=];
+    {{SUBMACHINE_EXTRA_REGISTERS}}
+
+    // We still need these registers for prover inputs.
+    reg query_arg_1;
+    reg query_arg_2;
+
+    std::machines::large_field::memory::Memory regs(byte2, MIN_DEGREE, LARGE_SUBMACHINES_MAX_DEGREE);
+    {{MEMORY_DECLARATION}}
+    {{SUBMACHINE_DECLARATIONS}}
+
+    let initial_memory: (fe, fe)[] = [
+        {{INITIAL_MEMORY}}
+    ];
+
+    function main {
```

Comment:
should `main` be the last thing in the file?

Reply:
done